src/components/Layout/Header/NavBar/UserMenu.tsx

```diff
@@ -118,10 +134,10 @@ const WalletButton: FC<WalletButtonProps> = ({
     () => (
       <HStack>
         {!(isConnected || isDemoWallet) && <WarningTwoIcon ml={2} w={3} h={3} color='yellow.500' />}
-        <WalletImage walletInfo={walletInfo} />
+        <WalletImage walletInfo={maybeMipdProvider?.info || walletInfo} />
```

Comment:
q: Same, I feel like the WalletProvider should provide those informations so we don't monkey patch everywhere, happy to discuss it, feasibility, maintainability etc

Reply:
I would prefer to keep detecting rdns in-place. WalletProvider is already bug-prone enough as-is and I wouldn't add more meta than necessary there: a `rdns` is all we need for normalized lookups, and tapping into the actual mipd store ensures that we get the right provider data and no race conditions/state corruptions: as long as the rdns used for lookup is the correct one, than the EIP6963/1993 provider will be the correct one.